internal/cri/server/container_status_test.go

```diff
@@ -302,3 +314,73 @@ func patchExceptedWithState(expected *runtime.ContainerStatus, state runtime.Con
 		expected.FinishedAt = 0
 	}
 }
+
+var _ containerd.Container = &fakeSpecOnlyContainer{}
+
+type fakeSpecOnlyContainer struct {
+	spec      *specs.Spec
+	errOnSpec error
+}
+
+// Spec implements client.Container.
+func (c *fakeSpecOnlyContainer) Spec(context.Context) (*specs.Spec, error) {
+	if c.errOnSpec != nil {
+		return nil, c.errOnSpec
+	}
+	return c.spec, nil
+}
+
+// Checkpoint implements client.Container.
+func (*fakeSpecOnlyContainer) Checkpoint(context.Context, string, ...containerd.CheckpointOpts) (containerd.Image, error) {
+	panic("unimplemented")
```

Comment:
nilt: Can we just result in a test error rather than panicking the test?

Reply:
Thanks, addressed in [cd8be1c](https://github.com/containerd/containerd/pull/9737/commits/cd8be1ccdd8609e66c67fadb0bab68b93b3231e0)